src/gmres_e.m

```diff
@@ -0,0 +1,426 @@
+function [x, flag, relresvec, time] = ...
+    gmres_e(A, b, m, k, tol, maxit, xInitial, varargin)
+    % GMRES-E algorithm
+    %
+    %   GMRES-E is a modified implementation of the restarted
+    %   Generalized Minimal Residual Error or GMRES(m) [1], performed by
+    %   appending 'k' eigenvectors corresponding to a few of the smallest
+    %   eigenvalues in magnitude for each outer iteration. In practice,
+    %   the approximate eigenvectors are the harmonic Ritz vectors
+    %   associated to the harmonic Ritz values per outer iteration.
+    %
+    %   Signature:
+    %   ----------
+    %
+    %   [x, flag, relresvec, time] = ...
+    %       lgmres(A, b, m, k, tol, maxit, xInitial)
+    %
+    %
+    %   Input Parameters:
+    %   -----------------
+    %
+    %   A:          n-by-n matrix
+    %               Left-hand side of the linear system Ax = b.
+    %
+    %   b:          n-by-1 vector
+    %               Right-hand side of the linear system Ax = b.
+    %
+    %   m:          int
+    %               Restart parameter (similar to 'restart' in MATLAB).
+    %
+    %   k:          int
```

Comment:
Same comment for `k`? Do we provide default values? 

Perhaps we can mention a suggested range based on the main reference. What do you think @gusespinola @jccf19 ?

Reply:
In [https://www.osti.gov/servlets/purl/15020368](url), page 6, the comment is that "Although full LGMRES is interesting from a theoretical point of view, it is not a practical algorithm. [...] ...optimal values of k are typically small, $k \leq 3$".

Reply:
According to the reference, "even just a few eigenvectors can make a big difference if the matrix has both small and large eigenvalues". However, I'm looking for more evidence on the upper bound for `k`.